ERCS/erc-xxxx.md

```diff
@@ -0,0 +1,243 @@
+If any of the validation or execution frames revers, returns an error object containing the revert message.
+Note that unlike `eth_call`, RIP-7560 transaction depends on its deployment and validation,
+so it can't be called in isolation from its deployment and validation
+If the transaction validation fails for any reason other than the failed signature check,
+returns an error object containing the details of the validation failure.
+
+Parameters:
+
+1. OBJECT - The RIP-7560 transaction object.
+   The `signature` field is optional.
+2. QUANTITY | TAG - integer block number, or the string "latest", "earliest", "pending", "safe" or "finalized"
+
+Returns:
+
+DATA - The return value of the `sender` execution frame.
+
+Error:
+
+DATA - The revert data of the first reverted frame.
+CODE - The error code indicating the type of error, which may include the entity that caused the revert on-chain.
+MESSAGE - The human-readable error that may include a decoding of the `DATA` field if possible.
+
+### Add RIP-7560 support for all remaining transaction-level RPC APIs
+
+This includes the following APIs: `eth_sendTransaction`, `eth_sendRawTransaction`,  `eth_getTransactionByHash`,
+`eth_getTransactionByBlockHashAndIndex`,
+`eth_getTransactionByBlockNumberAndIndex`.
+
+These methods have a very similar purpose and should support returning the new transaction type object.
+
+Note that the "transaction index position" is determined by the position of the transaction's **validation frame**.
+
+### Create a new JSON-RPC API method `eth_estimateRip7560TransactionGas`
+
+Performs a search for gas limit values required to make each of the frames of the RIP-7560 transaction execute
+successfully and without running out of gas.
+Note that for validation frames, only valid calls to an appropriate `AA_ENTRY_POINT` callback,
+such as `acceptAccount`, `acceptPaymaster`, `sigFailAccount` and `sigFailPaymaster`, is considered a success.
+
+If it fails to find such a value, returns an error message with the detailed description of the failure reason.
+
+Parameters:
+
+1. OBJECT - The RIP-7560 transaction object.
+   The `validationGasLimit`, paymasterValidationGasLimit, `paymasterGasLimit` and `callGasLimit` fields are optional.
+2. QUANTITY | TAG - integer block number, or the string "latest", "earliest", "pending", "safe" or "finalized"
+3. OBJECT - State override set
+   The State Override Set option allows you to change the state of a contract before executing the call. This means you
+   can modify the values of variables stored in the contract, such as balances and approvals for that call without
+   actually modifying the contract on the blockchain.
+   This behavior is equivalent to the one defined for the `eth_call` RPC method.
+
+Returns:
+
+1. Object
+
+| Name                        | Type     |
+|-----------------------------|----------|
+| validationGasLimit          | QUANTITY |
```

Comment:
returned validation values are not "limits", but actual values.
(because of our validation rules, there's no need to do "bisection" on validation values at all: if they can't access the current "glass ceiling"  gaslimit, they can't be affected by it, so we return actual gas, and that's enough gas gaslimit)

Reply:
Validation rules are not part of RIP-7562, not part of this ERC, and are generally opt-in. We cannot define core features of the protocol recklessly because there are magic rules that protect us from anything going wrong.